articles/calendar.perfplanet.org-2017.md

```diff
@@ -0,0 +1,170 @@
+# Pulling out all the stops in webfont performance
+
+Web fonts give you the awesome ability to provide text to your user in exactly the style that you want to. Unfortunately the default load behavior of web fonts is so bad, that they are one of the biggest performance detractors on the time to meaningful paint metric. After all, the user probably went to your site to read stuff, you should show them text as soon as possible.
+
+This problem has been described well in lots of conference talks and articles, and i you are new to this I recommend watching [Patrick Hamann](https://twitter.com/patrickhamann) in [The First Meaningful Paint](https://youtu.be/NGX0gYOo-Nk?t=18m59s) or reading [Ben Schwarz](https://twitter.com/benschwarz) in [The Critical Request](https://css-tricks.com/the-critical-request/)
+[70% of sites use web fonts](http://httparchive.org/interesting.php#fonts), where this quote is from:
+
+> [69% of sites use web fonts](http://httparchive.org/interesting.php#fonts), and unfortunately, they're providing a sub-par experience in most cases. They appear, then disappear, then appear again, change weights and jolt the page around during the render sequence.
+>
+> Frankly, this sucks on almost every level.
+> <footer>— <a href="https://css-tricks.com/the-critical-request/#article-header-id-4">Ben Schwarz, The Critical Request</a></footer>
+
+The behavior Ben is describing is what happens when web fonts load slowly, and trickle in gradually after the page has started rendering. The browser re-renders according to the [font style matching algorithm](https://www.w3.org/TR/css-fonts-3/#font-style-matching), which can cause normal text to suddenly be bold, or lighter, or jump around. A hot mess, and a really bad default for the best reading experience.
+
+Lets see what we can do about it!
+
+## Understanding performance detractors
+
+An HTTP request can generally be broken down into these phases which take up the bulk of the time until the request has finished:
+
+|| delay / queueing | DNS lookup | TLS handshake | TTFB | Download ||
+
+There might be a delay before the request even starts, you might have to look up the DNS for the domain, establish a secure connection, wait a bit for the server to respond to your request and finally wait for the download to finish.
+
+The default webfont loading happens like this:
+
+- HTML loads, the pre-scanner discovers some CSS
+- The CSS is queued for load, this is render blocking
+- When the CSS has loaded the page renders
+- The browser discovers that a webfont is needed to render some text
```

Comment:
Is it worth mentioning that the browser needs to construct the DOM and CSSOM to be able to figure this out? Even then, it might trigger webfont loading later, if the style isn't active right there and then.